src/common/object.ts

```diff
@@ -22,20 +22,35 @@
 type IsRequired = boolean;
 export type ValidateRegex = [Key, Value, RegExp, IsRequired?];
 
-export const validateRegex = (params: ValidateRegex[]): void => {
+export const validateRegex = <T extends ObjectLiteral>(
+  params: ValidateRegex[],
+  data?: T,
+): T => {
+  const mutatedData: ObjectLiteral = {
+    ...data,
+  };
   const result = params.reduce((result, [key, value, regex, isRequired]) => {
     if (isNullOrUndefined(value)) {
       return isRequired ? { ...result, [key]: `${key} is required!` } : result;
     }
 
-    const isValid = regex.test(value as string);
+    const matchResult = value.match(regex);
```

Comment:
@idoshamun do you have time to take a look, basically I think its false positive since `value`  is string which I match by regex and it is not constructed from user input its a static `handleRegex` value in `object.ts`?

Reply:
I see. We can safely ignore